communications/mqtt/publish/publish.ino

```diff
@@ -21,25 +21,31 @@ PubSubClient client(espClient);
 String jsonTopicStr;
 String newTopicStr;
 
+//Initialize time
+int before = millis();
 
 void setup() {
   Serial.begin(9600);  // Starts the serial communication
   Serial.println("");
 
   client.setServer(MQTT_BROKER_IP, MQTT_PORT);  // Connect the configured mqtt broker
-  //client.setCallback(callback);  // Prepare what to do when a message is recieved
 
   connectToWiFiNetwork();  // Connects to the configured network
   connectToMqttBroker();  // Connects to the configured mqtt broker
-  
 }
 
 void loop() {
+  //We check the connection every time
   checkConnections();
-  publishMqttData();  // Publishes to counter topic
-  publishMqttJson();  // Subscribes to json topic
   
-  delay(7000);
+  //But we publish information only every 5 seconds
+  int now = millis();
+  if (now-before >= 5000){
+    publishMqttData();  // Publishes to counter topic
+    publishMqttJson();  // Subscribes to json topic
+    before = millis();
```

Comment:
I would use `before = now;` to use exactly the same time

Reply:
But we want to have a difference between them of 5 seconds or so. If we just equal them there wont be any time difference...
If you mean to publish just when it's 5 seconds, aren't there a possibility that it may not happen that (now + 5000) is exactly equal to before ??

Reply:
be aware that every time you update the `now`. If you say that before and now, you would get 0>= 5000 and it wont be executed ;) thats the intention right?